### PR TITLE
Etching Tool

### PR DESCRIPTION
**Pull request by jalextowle**
# Description

This PR adds a debugging tool to the repo that can be used to debug failing or successful transactions on the Sepolia testnet. The debugger updates the file `test/debug/Debugging.t.sol` with the appropriate transaction values from a failing or successful transaction, and the test suite will create a Sepolia fork, etch the correct contracts onto the provided Hyperdrive instance, and run the transaction. The debugger can add logs to the Hyperdrive repo to investigate the failure and can change Hyperdrive's logic to verify a potential bug-fix.

image

h/t @jrhea for the dank meme

### CODE REVIEW

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -17,8 +17,9 @@ jobs:
     needs: detect-changes
     if: needs.detect-changes.outputs.changed == 'true'
     env:
-      GOERLI_RPC_URL: ${{ secrets.GOERLI_RPC_URL }}
       MAINNET_RPC_URL: ${{ secrets.MAINNET_RPC_URL }}
+      SEPOLIA_RPC_URL: ${{ secrets.SEPOLIA_RPC_URL }}
+      GOERLI_RPC_URL: ${{ secrets.GOERLI_RPC_URL }}
```
Comment (jrhea) [inline .github/workflows/benchmark.yml:22]: do we need goerli?

File: test/utils/EtchingUtils.sol [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+            vm.etch(address(hyperdrive), address(template).code);
+        }
+    }
+
+    function etchStETHHyperdrive(address _hyperdrive) internal {
+        // Get an interface to the target Hyperdrive instance. This will be
+        // used to load immutables that will be used during the etching process.
+        IHyperdrive hyperdrive = IHyperdrive(_hyperdrive);
+
+        // Etch the vault contract.        {
```
Comment (jrhea) [inline test/utils/EtchingUtils.sol:117]:
```suggestion
        // Etch the vault contract.
```

File: .github/workflows/benchmark.yml (hunk 1 above)
Comment (jalextowle) [reply .github/workflows/benchmark.yml]: Good call, we don't

**Review by jrhea — APPROVED** [review]
approved with one nit.  Another suggestion would be to add a readme in the debug folder. not a show stopper

**Review by mcclurejt — APPROVED** [review]
🚀

**Pull request merged**